src/functions/authoriser/framework/__tests__/createLogger.spec.ts

```diff
@@ -0,0 +1,112 @@
+import * as awsSdkMock from 'aws-sdk-mock';
+import * as sinon from 'sinon';
+
+import { uniqueLogStreamName, createLogger } from '../createLogger';
+import { CloudWatchLogs } from 'aws-sdk';
+
+describe('Logger', () => {
+
+  describe('uniqueLogStreamName', () => {
+    const sut = uniqueLogStreamName;
+
+    it('returns a string in the expected format', () => {
+      // ACT
+      const result = sut('LoggerName');
+
+      // ASSERT
+      expect(result).toMatch(/^LoggerName-\d\d\d\d-\d\d-\d\d-[0-9a-f]{32}$/);
+    });
+
+    it('generates unique names each time', () => {
+      const results = new Set();
+      const countToGenerate = 50000;
+
+      // ACT
+      for (let i = 0; i < countToGenerate; i = i + 1) {
+        const result = sut('LoggerName');
+        results.add(result);
+      }
+
+      // ASSERT
+      expect(results.size).toEqual(countToGenerate);
+    });
+  });
+
+  describe('createLogger', () => {
+
+    let createLogGroupSpy;
+    let createLogStreamSpy;
+    let putLogEventsSpy;
+
+    beforeEach(() => {
+
+      awsSdkMock.restore('CloudWatchLogs', 'createLogGroup');
+      awsSdkMock.restore('CloudWatchLogs', 'createLogStream');
+      awsSdkMock.restore('CloudWatchLogs', 'putLogEvents');
+
+      process.env.FAILED_LOGINS_CWLG_NAME = 'testLogGroupName';
+
+    });
+
+    it('should call the correct CloudWatchLogs methods', async () => {
+
+      createLogGroupSpy = sinon.stub().resolves(true);
```

Comment:
What's the reason for Sinon here? Typemoq seems to be prevalent on the authoriser, is there something that Sinon can accomplish that Typemoq cannot?

Reply:
`aws-sdk-mock` has a nice integration with sinon.